pkg/geoipupdate/database/http_reader.go

```diff
@@ -59,58 +60,122 @@ func (reader *HTTPDatabaseReader) Get(destination Writer, editionID string) erro
 		url.QueryEscape(destination.GetHash()),
 	)
 
-	req, err := http.NewRequest(http.MethodGet, maxMindURL, nil) // nolint: noctx
-	if err != nil {
-		return errors.Wrap(err, "error creating request")
-	}
-	req.SetBasicAuth(fmt.Sprintf("%d", reader.accountID), reader.licenseKey)
-
 	if reader.verbose {
 		log.Printf("Performing update request to %s", maxMindURL)
 	}
-	response, err := internal.MaybeRetryRequest(reader.client, reader.retryFor, req)
+
+	var modified bool
+	// It'd be nice to not use a temporary file here. However the Writer API does
+	// not currently support multiple attempts to download the file (it assumes
+	// we'll begin writing once). Using a temporary file here should be a less
+	// disruptive alternative to changing the API. If we change that API in the
+	// future, adding something like Reset() may be desirable.
```

Comment:
Potentially, we could check if the writer is an `io.WriteSeeker` and avoid the temporary file in most cases without changing the API. I think this sort of optimization is somewhat common in the Go standard libraries. That said, the temporary file probably isn't a big deal. I am slightly worried about someone not having the tempdir set up correctly in a weird container environment or something, but it probably won't be an issue.

Reply:
Hmm yeah, that's a good point. Although then we will have 2 code paths to test and such, so I'm a bit hesitant about going down that path if it's avoidable. I'm a little worried about the temporary file too but as you say, hopefully it will be ok. We can always adjust if necessary!